packet.go

```diff
@@ -154,6 +155,10 @@ func recvPacket(r io.Reader, alloc *allocator, orderID uint32) (uint8, []byte, e
 		debug("recv packet %d bytes too long", length)
 		return 0, nil, errLongPacket
 	}
+	if length < minMsgLength {
```

Comment:
RIght now, it would only have the one use case, but it might be a good helper function to have around regardless, for better clarity of purpose. Like rather than just having an `b = append(b, packetType)` we can have `b = marshalByte(b, packetType)` that gives semantic information about purpose.

But _that_ is a whole big can of bigger worms than just dealing with this. I think I’ll be behind any decision you come to here, the magic isn’t _that_ concerning… there is for instance, no sftp packet I am aware of that is just `packetLength + packeType` even though strictly that wouldn’t necessary be technically incompatible.

Reply:
Done. I did have to install a check for EOF below, to make sure that gets translated to errShortPacket.